crowdin/crowdin.go

```diff
@@ -270,107 +302,40 @@ func (c *Client) Delete(ctx context.Context, path string) (*Response, error) {
 	if r.StatusCode == http.StatusBadRequest {
-		respBody = &ValidationErrorResponse{Response: r, Status: r.StatusCode}
+		errorResponse = &model.ValidationErrorResponse{Response: r, Status: r.StatusCode}
 	} else {
-		respBody = &ErrorResponse{Response: r}
+		errorResponse = &model.ErrorResponse{Response: r}
 	}
 
-	data, err := io.ReadAll(r.Body)
-	if err == nil && len(data) > 0 {
-		err = json.Unmarshal(data, respBody)
-		if err != nil {
-			respBody = &ErrorResponse{
-				Response: r,
-				Err: Error{
-					Code:    r.StatusCode,
-					Message: err.Error(),
-				},
-			}
-		}
+	if err := json.Unmarshal(body, errorResponse); err != nil {
+		return fmt.Errorf("client: server returned %d status code", r.StatusCode)
 	}
-	return respBody
+	return errorResponse
 }
 
 // Response is a Crowdin response that wraps http.Response.
 type Response struct {
 	*http.Response
 
-	Pagination *Pagination
+	Pagination model.Pagination
 }
 
-func (r *Response) ParsePagination(body []byte) error {
-	if strings.HasPrefix(r.Header.Get("Content-Type"), "application/json") {
-		var p Pagination
-		if err := json.Unmarshal(body, &p); err != nil {
-			return err
-		}
-		r.Pagination = &p
+// populatePagination reads the pagination information from the response
+// body and sets it to the Response struct.
+func (r *Response) populatePagination(body []byte) error {
+	p := new(model.PaginationResponse)
+	if err := json.Unmarshal(body, p); err != nil {
+		return err
 	}
+	r.Pagination = p.Pagination
 	return nil
 }
 
-// Pagination represents the pagination information.
-type Pagination struct {
-	Offset int `json:"offset"`
-	Limit  int `json:"limit"`
-}
-
-type Error struct {
-	Code    int    `json:"code"`
-	Message string `json:"message"`
-}
-
-// ErrorResponse is the error response structure from the API.
-type ErrorResponse struct {
-	Response *http.Response `json:"-"`
-
-	Err Error `json:"error"`
-}
-
-// Error implements the Error interface.
-func (r *ErrorResponse) Error() string {
-	return fmt.Sprintf("%d %s", r.Err.Code, r.Err.Message)
-}
-
-type ValidationError struct {
-	Error struct {
-		Key    string `json:"key"`
-		Errors []struct {
-			Code    string `json:"code"`
-			Message string `json:"message"`
-		} `json:"errors"`
-	} `json:"error"`
-}
-
-// ValidationErrorResponse
-type ValidationErrorResponse struct {
-	Response *http.Response `json:"-"`
-
-	Errors []ValidationError `json:"errors"`
-	Status int
-}
-
-// Error implements the Error interface.
-func (r *ValidationErrorResponse) Error() string {
-	var sb strings.Builder
-	for i, err := range r.Errors {
-		if i != 0 {
-			sb.WriteString("; ")
-		}
-		sb.WriteString(fmt.Sprintf("%s: ", err.Error.Key))
-		for j, e := range err.Error.Errors {
-			if j != 0 {
-				sb.WriteString(", ")
-			}
-			sb.WriteString(fmt.Sprintf("%s (%s)", e.Message, e.Code))
-		}
-	}
-	return sb.String()
+// ToPtr is a helper function that returns a pointer
+// to the provided input.
+func ToPtr[T any](v T) *T {
```

Comment:
This does not seem to be used

Reply:
This helper can be useful on the client side to avoid writing unnecessary code when passing a pointer to a literal value.
For example, use
```
req := &model.BranchesMergeRequest{
    SourceBranchID: 1,
    DeleteAfterMerge: crowdin.ToPtr(true),
    DryRun: crowdin.ToPtr(false),
}
// crowdin.ToPtr(0)
```
instead of
```
deleleBranch := true
dryRyn := false
// intVal := 0

req = &model.BranchesMergeRequest{
	SourceBranchID: 1,
	DeleteAfterMerge: &deleleBranch,
	DryRun: &dryRyn,
}
```

Reply:
This does not look like the best DX. Can we simplify this?

Reply:
Do you mean `crowdin.ToPtr(...)` func or initializing structs in the examples?

Reply:
Yes. I don't think we should provide such helpers, it should be hidden from the developer or implemented on the integration side.

Is it possible to do the following?

```
req = &model.BranchesMergeRequest{
	SourceBranchID: 1,
	DeleteAfterMerge: false,
	DryRun: false,
}
```

Reply:
The short answer is No.
In Go, can't directly assign a value to a pointer variable. The reason we use pointers in this case is to allow for the possibility of a nil value. A `bool` in Go defaults to `false` (`int` defaults to `0`) if not initialized.
That is, if we don't pass the `DryRun` field, then nothing will be sent (instead of zero-value `false` in the case of no pointer).

Reply:
I think the helper func is a good practice that provides a convenient way to create pointers.
But if you think it's better to remove this helper, I'll make the necessary changes.